packages/date-picker/src/vaadin-date-picker-year-scroller.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * @license
+ * Copyright (c) 2016 - 2022 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import { css, registerStyles, ThemableMixin } from '@vaadin/vaadin-themable-mixin/vaadin-themable-mixin.js';
+import { InfiniteScroller } from './vaadin-infinite-scroller.js';
+
+registerStyles(
+  'vaadin-date-picker-year-scroller',
+  css`
+    :host {
+      --vaadin-infinite-scroller-item-height: 80px;
+      width: 50px;
+      display: block;
+      height: 100%;
+      position: absolute;
+      right: 0;
+      transform: translateX(100%);
+      -webkit-tap-highlight-color: transparent;
+      -webkit-user-select: none;
+      -moz-user-select: none;
+      user-select: none;
+      /* Center the year scroller position. */
+      --vaadin-infinite-scroller-buffer-offset: 50%;
+    }
+
+    :host::before {
+      content: '';
+      display: block;
+      background: transparent;
+      width: 0;
+      height: 0;
+      position: absolute;
+      left: 0;
+      top: 50%;
+      transform: translateY(-50%);
+      border-width: 6px;
+      border-style: solid;
+      border-color: transparent;
+      border-left-color: #000;
+    }
+  `,
+  { moduleId: 'vaadin-date-picker-year-scroller-styles' },
+);
+
+/**
+ * An element used internally by `<vaadin-date-picker>`. Not intended to be used separately.
+ *
+ * @extends InfiniteScroller
+ * @mixes ThemableMixin
+ * @private
+ */
+class DatePickerYearScroller extends ThemableMixin(InfiniteScroller) {
```

Comment:
I'm not sure whether we should make scrollers use `ThemableMixin` - in theory, corresponding styles could be still located in `vaadin-date-picker-overlay-content` instead. @jouni WDYT?